From dda55aa50db95a25f0d919c3a0d5871827cdc40f Mon Sep 17 00:00:00 2001
From: Petr Lautrbach <plautrba@redhat.com>
Date: Tue, 21 Jun 2016 15:15:00 +0200
Subject: [PATCH] framework: Use subprocess.check_output() with a sequence of
 program arguments

All subprocess.get*output() should be replaced with either
subprocess.Popen or subprocess.check_output now.

String args were converted to sequences of program arguments.
---
 framework/src/SetroubleshootFixit.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/framework/src/SetroubleshootFixit.py b/framework/src/SetroubleshootFixit.py
index a810151..aef1dd0 100644
--- a/framework/src/SetroubleshootFixit.py
+++ b/framework/src/SetroubleshootFixit.py
@@ -14,9 +14,9 @@ def __init__ (self, *p, **k):
         
     @dbus.service.method ("org.fedoraproject.SetroubleshootFixit", in_signature='ss', out_signature='s')
     def run_fix(self, local_id, analysis_id):
-        import commands
-        command = "sealert -f %s -P %s" % ( local_id, analysis_id)
-        return commands.getoutput(command)
+        import subprocess
+        command = ["sealert", "-f", local_id, "-P", analysis_id]
+        return subprocess.check_output(command, universal_newlines=True)
 
 if __name__ == "__main__":
     mainloop = gobject.MainLoop ()
